### PR TITLE
Fix #1956: Optimise the code in profile_chooser_add_view.xml

### LINKED ISSUE #1956: Optimise code in profile_chooser_add_view.xml
This issue is tracking the optimization work of `profile_chooser_add_view.xml`

1. Make sure that the layout uses `ConstraintLayout` and there is minimal nesting of views. If this view already uses `ConstraintLayout` then it's fine.

2. Make sure that all TextViews use the style from `styles.xml` file like `Heading1`, `TextFieldLabel`, `Body`, etc. whichever suits best.

While making the above changes the UI should not get affected and the end result should still be the same as before (or an optimized version of UI).

If there are multiple files with the same name, update them all.

## References
Mobile Portrait: https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/grid/
Mobile Landscape: https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/grid/
Tablet: https://xd.adobe.com/view/d405de00-a871-4f0f-73a0-f8acef30349b-a234/grid/
Wiki: https://github.com/oppia/oppia-android/wiki/Working-on-UI

### PR DESCRIPTION
**Pull request by Arjupta**
## Explanation

Fixes #1956 : Have optimised text views according to the style.xml and used constraint layout as the parent layout without affecting the original design 
## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by Arjupta** [comment]
@MohamedMedhat1998 PTAL

**Comment by BenHenning** [comment]
@ARJUPTA please update the title of the PR to match others (e.g. no space before the ``:``).

**Comment by Arjupta** [comment]
> @ARJUPTA please update the title of the PR to match others (e.g. no space before the `:`).

Is it okay now @BenHenning ?

**Comment by rt4914** [comment]
> > @ARJUPTA please update the title of the PR to match others (e.g. no space before the `:`).
> 
> Is it okay now @BenHenning ?

@ARJUPTA That looks correct now.

**Review by MohamedMedhat1998 — CHANGES_REQUESTED** [review]
The code change resulted in UI regressions in some cases:
1. Normal device - portrait mode - app first-start
Screenshot_2020-10-14-14-17-08
2. Normal device - portrait mode - app non-first-start
Screenshot_2020-10-14-14-13-35
Screenshot_2020-10-14-14-12-35
3. Tablet device - portrait mode - app non-first-start
screenshot-2020-10-14_14 21 35 374
4. Tablet device - landscape mode - app non-first-start
screenshot-2020-10-14_14 21 56 311

File: app/src/main/res/layout-land/profile_chooser_add_view.xml
Comment (ARJUPTA) [inline app/src/main/res/layout-land/profile_chooser_add_view.xml:41]: @MohamedMedhat1998 as I have not changed the properties related to the width of the Linear Layout then what can be the reason for the layout to expand. Is it because of constraining the ends to Start and End of Parent?

**Review by Arjupta — COMMENTED** [review]
@MohamedMedhat1998 needed help to recognise the error.
As I have not changed the properties related to the width of the Linear Layout then what can be the reason for the layout to expand. Is it because of constraining the ends to Start and End of Parent?

**Comment by MohamedMedhat1998** [comment]
> @MohamedMedhat1998 needed help to recognise the error.
> As I have not changed the properties related to the width of the Linear Layout then what can be the reason for the layout to expand. Is it because of constraining the ends to Start and End of Parent?

Try to take another look [here](https://github.com/oppia/oppia-android/pull/1999/files), you may find something that you changed by mistake. It can be something related to the margins, chain between views, padding, ...etc.  
I haven't taken a deeper look at your changes, but generally, they look good. Maybe you have missed something.  
I suggest that you enable `layout bounds` from the developer options, then change some values obviously and see how this will affect the UI.

File: app/src/main/res/layout-land/profile_chooser_add_view.xml
Comment (rt4914) [inline app/src/main/res/layout-land/profile_chooser_add_view.xml:32]: @ARJUPTA Maybe removal of this margin is creating the issues. You should not change the padding/margin in this PR.

**Review by rt4914 — COMMENTED** [review]
@ARJUPTA Have a look at my comment, it might help. (Not sure)

File: app/src/main/res/layout-land/profile_chooser_add_view.xml
Comment (Arjupta) [reply app/src/main/res/layout-land/profile_chooser_add_view.xml:32]: Thanks, but I think this is the only time I deleted a margin without replacement. As this file is going to be used for Landscape version then how is that the portrait mode design is also being affected
@rt4914

Comment (rt4914) [reply app/src/main/res/layout-land/profile_chooser_add_view.xml:32]: @ARJUPTA Yeah you are right. I haven't investigated the issue you are facing. I will defer to @MohamedMedhat1998 for that.

@MohamedMedhat1998 Can you review and provide update on this. And if needed then assign it to me.

**Comment by rt4914** [comment]
@ARJUPTA Did you try this https://github.com/oppia/oppia-android/pull/1999#issuecomment-708431291

**Comment by MohamedMedhat1998** [comment]
>     2\. Normal device - portrait mode - app non-first-start
Screenshot_2020-10-14-14-13-35

@ARJUPTA @rt4914 I was able to solve this one by changing line 45 from 
```
      app:layout_constraintTop_toBottomOf="@+id/add_profile_divider_view"
```
to
```
      app:layout_constraintTop_toTopOf="parent"
```
@ARJUPTA You can investigate by making random changes and observe the UI (make sure to turn on `layout bounds` while doing that.


But I had no luck with the following (I don't know why it isn't being fixed when it is on the RHS) @rt4914 any idea?
Screenshot_2020-10-14-14-12-35

**Comment by Arjupta** [comment]
Sorry for the late reply, I have tried the way you suggested @MohamedMedhat1998 , some were successful but only for certain configurations.

### **Landscape mode**
First or non-first start on mobile. This was achieved by removing the horizontal chain.
Screenshot_2020-10-19-16-05-44-64_943a62cb4c6fb83e010e1c2e82766a17

### Potrait Mode
Talking about mobile phones, there are three cases

**Case 1**
One or two profile added earlier, this does not cause any issue with layout 

**Case 2**
More than 2 profile added earlier, temporarily solved this by removing the bottom constraint of Outermost Linear Layout from the parent, thereby nullifying the effect of vertical chain. But this causes another problem that the No user added( for the first time) screen have a divider whose bottom margin does not work now. SS attatched after the below pic.

Screenshot_2020-10-19-16-05-28-65_943a62cb4c6fb83e010e1c2e82766a17

Screenshot_2020-10-19-17-36-02-45_943a62cb4c6fb83e010e1c2e82766a17
This maybe solved by increasing the marginTop value of LinearLayout below divider. It does have the match parent width problem but that is the part of **CASE 3**

**Case 3**
No profile added ie. for the first time making users. This is temporarily solved by contracting the width to 0dp, attaching the image to check is it the right layout

Screenshot_2020-10-19-15-46-29-58_943a62cb4c6fb83e010e1c2e82766a17

**Comment by Arjupta** [comment]
> > ```
> > 2\. Normal device - portrait mode - app non-first-start
> > ```
> 
> Screenshot_2020-10-14-14-13-35
> 
> @ARJUPTA @rt4914 I was able to solve this one by changing line 45 from
> 
> ```
>       app:layout_constraintTop_toBottomOf="@+id/add_profile_divider_view"
> ```
> 
> to
> 
> ```
>       app:layout_constraintTop_toTopOf="parent"
> ```
> 
> @ARJUPTA You can investigate by making random changes and observe the UI (make sure to turn on `layout bounds` while doing that.
> 
> But I had no luck with the following (I don't know why it isn't being fixed when it is on the RHS) @rt4914 any idea?
> Screenshot_2020-10-14-14-12-35

@MohamedMedhat1998 I read the official documentation about Constraint layout and found that this is caused due to the divider which is **GONE** in this RHS case. Maybe this will help you

**Comment by MohamedMedhat1998** [comment]
> @MohamedMedhat1998 I read the official documentation about Constraint layout and found that this is caused due to the divider which is **GONE** in this RHS case. Maybe this will help you

This divider is **GONE** in the left hand side too and it is working as expected, but I don't know why it isn't working in the right hand side.

**Comment by Arjupta** [comment]
> > @MohamedMedhat1998 I read the official documentation about Constraint layout and found that this is caused due to the divider which is **GONE** in this RHS case. Maybe this will help you
> 
> This divider is **GONE** in the left hand side too and it is working as expected, but I don't know why it isn't working in the right hand side.

@MohamedMedhat1998 can it be possible that this is caused by the parent layout with which this layout is associated. It might be because of any property like gravity of parent layout which is visible when there is another profile view beside it.

**Comment by rt4914** [comment]
@MohamedMedhat1998 @ARJUPTA PTAL

It was quite weird but seems like I have finally found the solution.

I have only focused on the mobile-portrait xml file.

You can find the solution here: https://gist.github.com/rt4914/01e113e4736011f5e1d8b171ba9def2a

Some thoughts:
1. If I removed `app:layout_constraintBottom_toBottomOf` from LinearLayout which solved the issue of extra top space in case of multiple profiles.
2. I have introduced `Space` instead of `marginBottom=60dp` because in my personal experience marginBottom does not work correctly a lot of times. (Don't know the exact reason but I have faced this multiple times.)

**Comment by Arjupta** [comment]
> @MohamedMedhat1998 @ARJUPTA PTAL
> 
> It was quite weird but seems like I have finally found the solution.
> 
> I have only focused on the mobile-portrait xml file.
> 
> You can find the solution here: https://gist.github.com/rt4914/01e113e4736011f5e1d8b171ba9def2a
> 
> Some thoughts:
> 
> 1. If I removed `app:layout_constraintBottom_toBottomOf` from LinearLayout which solved the issue of extra top space in case of multiple profiles.
> 2. I have introduced `Space` instead of `marginBottom=60dp` because in my personal experience marginBottom does not work correctly a lot of times. (Don't know the exact reason but I have faced this multiple times.)

@rt4914 Saw your gist. That margin-bottom doesn't work because there is vertical chain applied to constraint layout, which needs child to be constrained along that axis. so removing **app:layout_constraintBottom_toBottom** causes it.

**Comment by rt4914** [comment]
> > @MohamedMedhat1998 @ARJUPTA PTAL
> > It was quite weird but seems like I have finally found the solution.
> > I have only focused on the mobile-portrait xml file.
> > You can find the solution here: https://gist.github.com/rt4914/01e113e4736011f5e1d8b171ba9def2a
> > Some thoughts:
> > 
> > 1. If I removed `app:layout_constraintBottom_toBottomOf` from LinearLayout which solved the issue of extra top space in case of multiple profiles.
> > 2. I have introduced `Space` instead of `marginBottom=60dp` because in my personal experience marginBottom does not work correctly a lot of times. (Don't know the exact reason but I have faced this multiple times.)
> 
> @rt4914 Saw your gist. That margin-bottom doesn't work because there is vertical chain applied to constraint layout, which needs child to be constrained along that axis. so removing **app:layout_constraintBottom_toBottom** causes it.

@ARJUPTA I am not able to understand what do you mean here? Does this work for you or not because seems to work for me correctly.

**Comment by Arjupta** [comment]
> > > @MohamedMedhat1998 @ARJUPTA PTAL
> > > It was quite weird but seems like I have finally found the solution.
> > > I have only focused on the mobile-portrait xml file.
> > > You can find the solution here: https://gist.github.com/rt4914/01e113e4736011f5e1d8b171ba9def2a
> > > Some thoughts:
> > > 
> > > 1. If I removed `app:layout_constraintBottom_toBottomOf` from LinearLayout which solved the issue of extra top space in case of multiple profiles.
> > > 2. I have introduced `Space` instead of `marginBottom=60dp` because in my personal experience marginBottom does not work correctly a lot of times. (Don't know the exact reason but I have faced this multiple times.)
> > 
> > 
> > @rt4914 Saw your gist. That margin-bottom doesn't work because there is vertical chain applied to constraint layout, which needs child to be constrained along that axis. so removing **app:layout_constraintBottom_toBottom** causes it.
> 
> @ARJUPTA I am not able to understand what do you mean here? Does this work for you or not because seems to work for me correctly.

I just meant to say, that the problem you mentioned in thoughts is probably because of not using 
`app:layout_constraintBottom_toBottom`
This is because any constraint layout which has vertical chaining, needs its child views to be constrained to each other vertically and also the first/last child views needs to be constrained to parent

**Comment by rt4914** [comment]
> > > > @MohamedMedhat1998 @ARJUPTA PTAL
> > > > It was quite weird but seems like I have finally found the solution.
> > > > I have only focused on the mobile-portrait xml file.
> > > > You can find the solution here: https://gist.github.com/rt4914/01e113e4736011f5e1d8b171ba9def2a
> > > > Some thoughts:
> > > > 
> > > > 1. If I removed `app:layout_constraintBottom_toBottomOf` from LinearLayout which solved the issue of extra top space in case of multiple profiles.
> > > > 2. I have introduced `Space` instead of `marginBottom=60dp` because in my personal experience marginBottom does not work correctly a lot of times. (Don't know the exact reason but I have faced this multiple times.)
> > > 
> > > 
> > > @rt4914 Saw your gist. That margin-bottom doesn't work because there is vertical chain applied to constraint layout, which needs child to be constrained along that axis. so removing **app:layout_constraintBottom_toBottom** causes it.
> > 
> > 
> > @ARJUPTA I am not able to understand what do you mean here? Does this work for you or not because seems to work for me correctly.
> 
> I just meant to say, that the problem you mentioned in thoughts is probably because of not using
> `app:layout_constraintBottom_toBottom`
> This is because any constraint layout which has vertical chaining, needs its child views to be constrained to each other vertically and also the first/last child views needs to be constrained to parent

@ARJUPTA Have you tried the solution that I have mentioned?

File: app/src/main/res/layout/profile_chooser_add_view.xml
Comment (rt4914) [inline app/src/main/res/layout/profile_chooser_add_view.xml]: EOF is missing, add one blank line at the end.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@ARJUPTA 
Great work.

There is just one issue left.

If you run this in tablet, this UI is
Screenshot_1604410154

PTAL.

File: app/src/main/res/layout/profile_chooser_add_view.xml
Comment (Arjupta) [reply app/src/main/res/layout/profile_chooser_add_view.xml]: Done , Pardon me for the delay

**Review by rt4914 — APPROVED** [review]
Thanks @ARJUPTA Will need approval from other reviewers too.

**Comment by Arjupta** [comment]
@MohamedMedhat1998 PTAL

**Review by MohamedMedhat1998 — CHANGES_REQUESTED** [review]
There are some required changes which I believe won't take too much time.
1. In tablet screens, there should be some **horizontal margins** like those we have in the profile_view  
screenshot-2020-11-16_14 19 40 657
screenshot-2020-11-16_14 19 24 752
2. In the mobile screens, the add_view has an extra top margin, but when I rotate the device twice, this extra margin disappears
Screenshot_2020-11-16-12-14-16

**Comment by Arjupta** [comment]
> There are some required changes which I believe won't take too much time.
> 
> 1. In tablet screens, there should be some **horizontal margins** like those we have in the profile_view
>    screenshot-2020-11-16_14 19 40 657
>    screenshot-2020-11-16_14 19 24 752


The code already contains the margin property
`     app:layoutMarginEnd="@{hasProfileEverBeenAddedValue ? @dimen/space_0dp : @dimen/space_24dp}"
      app:layoutMarginStart="@{hasProfileEverBeenAddedValue ? @dimen/space_0dp : @dimen/space_24dp}"
 `

@MohamedMedhat1998 we can achieve this only if we define layout margins to be present everytime not depending on the hasProfileEverBeenAddedValue property. Are you suggesting this? This may cause some faults in the layout after some profiles are added other than admin.

**Comment by Arjupta** [comment]
> 2. In the mobile screens, the add_view has an extra top margin, but when I rotate the device twice, this extra margin disappears
>    Screenshot_2020-11-16-12-14-16

@MohamedMedhat1998 This issue is not reproducible for me. I tried this after building the apk from the last commit on this PR. Please check if you did the same.

**Comment by MohamedMedhat1998** [comment]
> There are some required changes which I believe won't take too much time.
> 
>     1. In tablet screens, there should be some **horizontal margins** like those we have in the profile_view
>        screenshot-2020-11-16_14 19 40 657
>        screenshot-2020-11-16_14 19 24 752
> 
>     2. In the mobile screens, the add_view has an extra top margin, but when I rotate the device twice, this extra margin disappears
>        Screenshot_2020-11-16-12-14-16

@ARJUPTA I see that the margins exist in the code but it contains regression at my side.  
@rt4914 @anandwana001 can someone test this on his side, please?

**Comment by rt4914** [comment]
> > There are some required changes which I believe won't take too much time.
> > ```
> > 1. In tablet screens, there should be some **horizontal margins** like those we have in the profile_view
> >    screenshot-2020-11-16_14 19 40 657
> >    screenshot-2020-11-16_14 19 24 752
> > 
> > 2. In the mobile screens, the add_view has an extra top margin, but when I rotate the device twice, this extra margin disappears
> >    Screenshot_2020-11-16-12-14-16
> > ```
> 
> @ARJUPTA I see that the margins exist in the code but it contains regression at my side.
> @rt4914 @anandwana001 can someone test this on his side, please?

@ARJUPTA @MohamedMedhat1998 

All the mocks look correct to me tablet and mobile (portrait + landscape).

Screenshot_1605610682
Screenshot_1605610686
Screenshot_1605610792
Screenshot_1605610796
Screenshot_1605610871
Screenshot_1605610876
Screenshot_1605610906
Screenshot_1605610911

**Comment by MohamedMedhat1998** [comment]
> 
> 
> > > There are some required changes which I believe won't take too much time.
> > > ```
> > > 1. In tablet screens, there should be some **horizontal margins** like those we have in the profile_view
> > >    screenshot-2020-11-16_14 19 40 657
> > >    screenshot-2020-11-16_14 19 24 752
> > > 
> > > 2. In the mobile screens, the add_view has an extra top margin, but when I rotate the device twice, this extra margin disappears
> > >    Screenshot_2020-11-16-12-14-16
> > > ```
> > 
> > 
> > @ARJUPTA I see that the margins exist in the code but it contains regression at my side.
> > @rt4914 @anandwana001 can someone test this on his side, please?
> 
> @ARJUPTA @MohamedMedhat1998
> 
> All the mocks look correct to me tablet and mobile (portrait + landscape).
> 
> Screenshot_1605610682
> Screenshot_1605610686
> Screenshot_1605610792
> Screenshot_1605610796
> Screenshot_1605610871
> Screenshot_1605610876
> Screenshot_1605610906
> Screenshot_1605610911

@rt4914 photos numbers 5, 6, 7 contain regression, which is the "add_view" icon isn't aligned with the "profile_view" icon.

**Comment by rt4914** [comment]
> > > > There are some required changes which I believe won't take too much time.
> > > > ```
> > > > 1. In tablet screens, there should be some **horizontal margins** like those we have in the profile_view
> > > >    screenshot-2020-11-16_14 19 40 657
> > > >    screenshot-2020-11-16_14 19 24 752
> > > > 
> > > > 2. In the mobile screens, the add_view has an extra top margin, but when I rotate the device twice, this extra margin disappears
> > > >    Screenshot_2020-11-16-12-14-16
> > > > ```
> > > 
> > > 
> > > @ARJUPTA I see that the margins exist in the code but it contains regression at my side.
> > > @rt4914 @anandwana001 can someone test this on his side, please?
> > 
> > 
> > @ARJUPTA @MohamedMedhat1998
> > All the mocks look correct to me tablet and mobile (portrait + landscape).
> > Screenshot_1605610682
> > Screenshot_1605610686
> > Screenshot_1605610792
> > Screenshot_1605610796
> > Screenshot_1605610871
> > Screenshot_1605610876
> > Screenshot_1605610906
> > Screenshot_1605610911
> 
> @rt4914 photos numbers 5, 6, 7 contain regression, which is the "add_view" icon isn't aligned with the "profile_view" icon.

@MohamedMedhat1998 Yeah, makes sense to me. Thanks.

**Comment by Arjupta** [comment]
@MohamedMedhat1998 as it is confirmed that the layout doesn't show margin even after there exists layout margin properties for it. Is it because of any wrong code use of hasProfileAlreadyAdded property? What should I do

**Comment by anandwana001** [comment]
unassigning myself as mohammed is taking care of this issue.

**Comment by MohamedMedhat1998** [comment]
> @MohamedMedhat1998 as it is confirmed that the layout doesn't show margin even after there exists layout margin properties for it. Is it because of any wrong code use of hasProfileAlreadyAdded property? What should I do

@ARJUPTA I really don't know what is causing the problem here, but I suggest that you use the `space` view as you have done in line 33 in the `sw600dp-land` file.  
```
    <Space
      android:id="@+id/blank_space"
      android:layout_width="the correct value"
      android:layout_height="match_parent"
      android:visibility="@{hasProfileEverBeenAddedValue ? View.GONE : View.VISIBLE}"
      app:layout_constraintEnd_toEndOf="parent"
      app:layout_constraintStart_toStartOf="parent"
      app:layout_constraintTop_toBottomOf="@id/add_profile_divider_view" />
```
@rt4914 will this be a valid approach?

**Comment by rt4914** [comment]
> > @MohamedMedhat1998 as it is confirmed that the layout doesn't show margin even after there exists layout margin properties for it. Is it because of any wrong code use of hasProfileAlreadyAdded property? What should I do
> 
> @ARJUPTA I really don't know what is causing the problem here, but I suggest that you use the `space` view as you have done in line 33 in the `sw600dp-land` file.
> 
> ```
>     <Space
>       android:id="@+id/blank_space"
>       android:layout_width="the correct value"
>       android:layout_height="match_parent"
>       android:visibility="@{hasProfileEverBeenAddedValue ? View.GONE : View.VISIBLE}"
>       app:layout_constraintEnd_toEndOf="parent"
>       app:layout_constraintStart_toStartOf="parent"
>       app:layout_constraintTop_toBottomOf="@id/add_profile_divider_view" />
> ```
> 
> @rt4914 will this be a valid approach?

@MohamedMedhat1998 This can work.

**Comment by Arjupta** [comment]
@MohamedMedhat1998 @rt4914 I have tried to use space view for handling margin issues but the project gives error on building
The new code is as follows - 
    
    <?xml version="1.0" encoding="utf-8"?>
    
    <layout xmlns:android="http://schemas.android.com/apk/res/android" xmlns:app="http://schemas.android.com/apk/res-auto">

    <data>

    <import type="android.widget.LinearLayout" />

    <import type="android.view.Gravity" />

    <import type="android.view.View" />

    <variable
      name="hasProfileEverBeenAddedValue"
      type="androidx.databinding.ObservableField&lt;Boolean&gt;" />
    </data>

    <androidx.constraintlayout.widget.ConstraintLayout
      android:layout_width="match_parent"
      android:layout_height="wrap_content"
      app:layout_constraintVertical_chainStyle="packed">

    <View
      android:id="@+id/add_profile_divider_view"
      android:layout_width="match_parent"
      android:layout_height="0.5dp"
      android:background="@color/oppiaProfileChooserDivider"
      android:visibility="@{hasProfileEverBeenAddedValue ? View.GONE : View.VISIBLE}"
      app:layout_constraintEnd_toEndOf="parent"
      app:layout_constraintStart_toStartOf="parent"
      app:layout_constraintTop_toTopOf="parent" />

    <Space
      android:id="@+id/blank_space"
      android:layout_width="match_parent"
      android:layout_height="60dp"
      android:visibility="@{hasProfileEverBeenAddedValue ? View.GONE : View.VISIBLE}"
      app:layout_constraintEnd_toEndOf="parent"
      app:layout_constraintStart_toStartOf="parent"
      app:layout_constraintTop_toBottomOf="@id/add_profile_divider_view" />

    <LinearLayout
      android:id="@+id/add_profile_item"
      android:layout_width="match_parent"
      android:layout_height="wrap_content"
      android:gravity="@{hasProfileEverBeenAddedValue ? Gravity.CENTER_HORIZONTAL : Gravity.CENTER_VERTICAL}"
      android:orientation="@{hasProfileEverBeenAddedValue ? LinearLayout.VERTICAL : LinearLayout.HORIZONTAL}"
      app:layout_constraintEnd_toEndOf="parent"
      app:layout_constraintStart_toStartOf="parent"
      app:layout_constraintTop_toBottomOf="@+id/blank_space"
      app:layoutMarginTop="@{hasProfileEverBeenAddedValue ? @dimen/space_0dp : @dimen/profile_chooser_add_view_margin_top_profile_not_added}">

      <Space
        android:id="@+id/margin_start"
        android:layout_width="@{hasProfileEverBeenAddedValue ? @dimen/profile_chooser_add_view_margin_start_profile_already_added : @dimen/profile_view_not_added_margin}"
        android:layout_height="match_parent"/>

      <de.hdodenhof.circleimageview.CircleImageView
        android:id="@+id/profile_add_button"
        android:layout_width="72dp"
        android:layout_height="72dp"
        android:src="@drawable/ic_add_profile"
        app:civ_border_color="@color/avatarBorder"
        app:civ_border_width="1dp" />

      <LinearLayout
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:gravity="@{hasProfileEverBeenAddedValue ? Gravity.CENTER_HORIZONTAL : Gravity.CENTER_VERTICAL}"
        android:orientation="vertical"
        app:layoutMarginTop="@{hasProfileEverBeenAddedValue ? @dimen/profile_view_already_added_description_parent_margin_top : @dimen/space_0dp}"
        app:layoutMarginStart="@{hasProfileEverBeenAddedValue ? @dimen/space_0dp : @dimen/profile_chooser_add_view_description_parent_margin_start_profile_not_added}">

        <TextView
          android:id="@+id/add_profile_text"
          style="@style/Caption"
          android:text="@{hasProfileEverBeenAddedValue ? @string/profile_chooser_add : @string/set_up_multiple_profiles}"
          android:textColor="@color/white" />

        <TextView
          android:id="@+id/add_profile_description_text"
          style="@style/TextFieldLabel"
          android:layout_gravity="center"
          android:layout_marginTop="4dp"
          android:text="@string/set_up_multiple_profiles_description"
          android:textColor="@color/white"
          android:visibility="@{hasProfileEverBeenAddedValue ? View.GONE : View.VISIBLE}" />
      </LinearLayout>

      <Space
        android:id="@+id/margin_end"
        android:layout_width="@{hasProfileEverBeenAddedValue ? @dimen/profile_chooser_add_view_margin_end_profile_already_added : @dimen/profile_view_not_added_margin}"
        android:layout_height="match_parent"/>

    </LinearLayout>
    </androidx.constraintlayout.widget.ConstraintLayout>
    </layout>


The build fails at this error - **Cannot find a setter for <android.widget.Space android:layout_width> that accepts parameter type 'float' . If a binding adapter provides the setter, check that the adapter is annotated correctly and that the parameter type matches.**

**Comment by rt4914** [comment]
https://github.com/oppia/oppia-android/pull/2163 and this PR both are getting a bit hard to solve. I will need to try it to find the solution. Mostly I will be able to do this after a week or so.

**Comment by Arjupta** [comment]
As discussed this PR will now be closed, a new PR will be created by @rt4914 to resolve this issue later on.

**Pull request closed without merging**
